projectq/backends/_azure/_azure_quantum.py

```diff
@@ -0,0 +1,403 @@
+            state (str): A state in bit-string format.
+            qureg (Qureg): A ProjectQ Qureg object.
+
+        Returns:
+            float: The probability for the provided state.
+        """
+        if len(state) != len(qureg):
+            raise ValueError('Desired state and register must be the same length!')
+
+        probs = self.get_probabilities(qureg)
+
+        try:
+            return probs[state]
+        except KeyError:
+            return 0.0
+
+    def get_probabilities(self, qureg):
+        """
+        Given the provided qubit register, determine the probability of each possible outcome.
+
+        Note:
+            This method should only be called *after* a circuit has been run and its results are available.
+
+        Args:
+            qureg (Qureg): A ProjectQ Qureg object.
+
+        Returns:
+            dict: A dict mapping of states -> probability.
+        """
+        if len(self._probabilities) == 0:
+            raise RuntimeError("Please, run the circuit first!")
+
+        probability_dict = {}
+        for state in self._probabilities:
+            mapped_state = ['0'] * len(qureg)
+            for i, qubit in enumerate(qureg):
+                try:
+                    meas_idx = self._measured_ids.index(qubit.id)
+                except ValueError:
+                    continue
+                mapped_state[i] = state[meas_idx]
+            probability = self._probabilities[state]
+            mapped_state = "".join(mapped_state)
+            probability_dict[mapped_state] = probability_dict.get(mapped_state, 0) + probability
+        return probability_dict
+
+    @property
+    def _input_data(self):
+        qubit_mapping = self.main_engine.mapper.current_mapping
+        qubits = len(qubit_mapping.keys())
+
+        if self._provider_id == IONQ_PROVIDER_ID:
+            return {
+                "qubits": qubits,
+                "circuit": self._circuit
+            }
+        elif self._provider_id == QUANTINUUM_PROVIDER_ID:
+            measurement_gates = ""
+
+            for measured_id in self._measured_ids:
+                qb_loc = self.main_engine.mapper.current_mapping[measured_id]
+                measurement_gates += "measure q[{0}] -> c[{0}];\n".format(qb_loc)
+
+            return f"OPENQASM 2.0;\ninclude \"qelib1.inc\";\nqreg q[{qubits}];\ncreg c[{qubits}];" \
+                   f"{self._circuit}\n{measurement_gates}"
+
+    @property
+    def _metadata(self):
+        qubit_mapping = self.main_engine.mapper.current_mapping
+        num_qubits = len(qubit_mapping.keys())
+        meas_map = [qubit_mapping[qubit_id] for qubit_id in self._measured_ids]
+
+        return {
+            "num_qubits": num_qubits,
+            "meas_map": meas_map
+        }
+
+    def estimate_cost(self, **kwargs):
+        """Estimate cost for the circuit this object has built during engine execution."""
+        return self._target.estimate_cost(
+            circuit=self._input_data,  # noqa
```

Comment:
Please specify the warning you are disabling explicitly (e.g. `# noqa: F401`)

Reply:
I removed most of `noqa` comments, initially I followed IDE suggestions, but after running `pre-commit` it became clear where noqa actually needed. 

This applies for rest of noqa comments as well.